app/models/category.rb

```diff
@@ -2,9 +2,8 @@ class Category < ApplicationRecord
   has_many :products, dependent: :destroy
   has_many :children, class_name: Category.name, foreign_key: :parent_id
   belongs_to :parent, class_name: Category.name
+  validates :title, presence: true
 
   scope :parent_category, ->{where parent_id: nil}
```

Comment:
em không hiểu tai sao khi em sửa children_category thành children thì không sao. Nhưng khi em sửa parent_category thành parent thì nó lại báo lỗi : " You tried to define a scope named "parent" on the model "Category", but Active Record already defined a class method with the same name."